glommio/src/io/dma_file.rs

```diff
@@ -221,51 +229,61 @@ impl DmaFile {
         Ok(ReadResult::from_whole_buffer(buffer))
     }
 
-    /// Issues fdatasync into the underlying file.
+    /// Issues `fdatasync` for the underlying file, instructing the OS to flush
+    /// all reads/writes to the device, providing durability even if the system
+    /// crashes or is rebooted.
+    ///
+    /// As this is a DMA file, the OS should not be caching this file; however,
```

Comment:
it is guaranteed that the OS should not be caching the file so there is no concern there.

The reason `fdatasync` is needed is, as you well put later, that the device also has an internal cache.